engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java

```diff
@@ -973,6 +973,93 @@ public void shouldTerminateSequentialMultiInstanceElement() {
             tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATED));
   }
 
+  @Test
+  public void shouldTerminateMultiInstanceBodyAndNestedElements() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .subProcess(
+                    "A",
+                    s ->
+                        s.embeddedSubProcess()
+                            .startEvent()
+                            .userTask("B")
+                            .endEvent()
+                            .subProcessDone()
+                            .multiInstance()
+                            .parallel()
+                            .zeebeInputCollectionExpression("[1,2,3]")
+                            .multiInstanceDone())
+                .endEvent()
+                .done())
+        .deploy();
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var multiInstanceElement =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId("A")
+            .withElementType(BpmnElementType.MULTI_INSTANCE_BODY)
+            .getFirst();
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("B")
+        .withElementType(BpmnElementType.USER_TASK)
+        .limit(3)
+        .map(Record::getKey)
+        .toList();
```

Comment:
:wrench: We could assert the number of activated elements to make it clear that we want to wait for the user tasks.

Currently, we extract the keys of the records but don't use them. Instead, we use the keys of the subprocess instances.